scala/scala.bzl

```diff
@@ -472,6 +472,17 @@ def dep_target_contains_ijar(dep_target):
   return (hasattr(dep_target, 'scala') and hasattr(dep_target.scala, 'outputs') and
           hasattr(dep_target.scala.outputs, 'ijar') and dep_target.scala.outputs.ijar)
 
+# When import mavan_jar's for scala macros we have to use the jar:file requirement
+# since bazel 0.6.0 this brings in the source jar too
+# the scala compiler thinks a source jar can look like a package space
+# causing a conflict between objects and packages warning
+#  error: package cats contains object and package with same name: implicits
+# one of them needs to be removed from classpath
+# import cats.implicits._
+
+def filter_not_sources(deps):
+  return depset([dep for dep in deps.to_list() if "-sources.jar" not in dep.basename ])
```

Comment:
Might sound picky but any chance you can factor out the if so that it's shared between the two places?

Reply:
yeah maybe:

```python
def _is_source_jar(filename):
  return "-sources.jar" in filename
```
then we call that twice? `if not _is_source_jar(dep.basename)`?